core/arns/src/main/java/software/amazon/awssdk/arns/Arn.java

```diff
@@ -138,6 +138,70 @@ public static Builder builder() {
         return new DefaultBuilder();
     }
 
+    /**
+     * Attempts to parse the given string into an {@link Arn}. If the input string is not a valid ARN,
+     * this method returns {@link Optional#empty()} instead of throwing an exception.
+     * <p>
+     * When successful, the resource is accessible entirely as a string through
+     * {@link #resourceAsString()}. Where correctly formatted, a parsed resource
+     * containing resource type, resource and qualifier is available through
+     * {@link #resource()}.
+     *
+     * @param arn A string containing an ARN to parse.
+     * @return An {@link Optional} containing the parsed {@link Arn} if valid, or empty if invalid.
+     */
+    public static Optional<Arn> tryFromString(String arn) {
```

Comment:
It contains a lot of duplicate code. Can we extract common code?

One idea is to create a helper method that takes a boolean for throwing behavior `private static Optional<Arn> parseArn(String arn, boolean throwOnError) {...}`